serenity-report-resources/src/main/resources/freemarker/default.ftl

```diff
@@ -332,24 +332,24 @@
                 <#macro write_step(step, step_number)>
                 <@step_details step=step step_number=step_number level=level/>
                 <#if step.isAGroup()>
-                <#if level == 1>
                 <tr>
                 <td colspan="5">
                 <table id="stepSection${step_number}" style="display:none; width:100%">
 
-                </#if>
                 <#assign level = level + 1>
+                <#assign substep_number = 1>
                 <#list step.children as nestedStep>
-                    <@write_step step=nestedStep step_number=""/>
+                    <#if step.isAGroup() >
+                        <#assign substep_number = substep_number + 1>
+                        <@write_step step=nestedStep step_number=step_number + "-" + substep_number/>
```

Comment:
Use the step number in conjunction with the substep number to obtain a unique ID for the sub step